### PR TITLE
FIX: Indexing

### LINKED ISSUE #13: Dropped epochs
@larsoner: for 925b `in_names_match` is `["ba", "wa", "tone"]`. In fact that should never happen. 
Also, for the IDS paradigm rejecting sentence ERFs based on amplitude thresholds will be problematic

[mnefun] mnefun/_epoching.py (Line 223)

```
 e.equalize_event_counts(in_names_match)
```

> Traceback (most recent call last):
>   File "/home/ktavabi/Github/badbaby/badbaby/processing/run_mnefun.py", line 101, in <module>
>     mnefun.do_processing(
>   File "/home/ktavabi/Github/mnefun/mnefun/_mnefun.py", line 524, in do_processing
>     outs[ii] = func(p, subjects, p.in_names, p.in_numbers,
>   File "/home/ktavabi/Github/mnefun/mnefun/_epoching.py", line 223, in save_epochs
>     e.equalize_event_counts(in_names_match)
>   File "/home/ktavabi/Github/mne-python/mne/epochs.py", line 1947, in equalize_event_counts
>     elif not event_ids:
> ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

[Open in IDE](https://api.codestream.com/c/YBtNEotjDWL9G1o4/QXeE5Vy7SV6oculvb4axCA?marker=608b962cd7a92659f3a9d8f2&ide=default&src=GitHub) · [Open on GitHub](https://github.com/labsn/mnefun/blob/7138f44c2eaf6beb5616e0c14d728dc9f151b747/mnefun/_epoching.py#L223)

<sup>Created from VS Code using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=issue&utm_campaign=github)</sup>

### PR DESCRIPTION
**Pull request by larsoner**
Closes #13

Needs https://github.com/LABSN/mnefun/pull/339

On these two branches I see:
```
Doing epoch EQ/DQ. 
  Loading raw files for subject bad_925b.
    Epoching data (decim=5 -> sfreq=360.0 Hz).
    Computing autoreject thresholds: grad=3926 fT/cm
    Events restricted to those in params.in_numbers
    Matching trial counts and saving data to disk.
      Analysis "All": 432 epochs / condition
        Equalizing: ['std', 'ba', 'wa']
      Analysis "Individual": 19 epochs / condition
        Equalizing: ['std', 'ba']
      Analysis "Oddball": 27, 46 epochs / condition
      Analysis "AM": 107 epochs / condition
        Got 0 epochs for: ['s3', 's5']
      Analysis "IDs": 0, 1 epochs / condition
    Saving epochs to disk.
  (0:00:21.869)
Done
Successfully processed 1/1, bad:
[]

```
I also delete `push_raw` and `fetch_sss` steps since they should never need to be run (it's just used for remote MaxFilter steps, which this pipeline doesn't use)

### CODE REVIEW

**Comment by larsoner** [comment]
Also @ktavabi in principle we should probably use custom trial-count equalization for the `oddball` case, since the order of operations is equalize-then-combine, this is why we see " Analysis "Oddball": 27, 46 epochs / condition" (standard ends up with fewer than oddball). With a custom trial-count-eq, we can equalize ba and wa, then collapse, then equalize std and dev. Want me to implement that here, too, or in a separate PR?

**Comment by larsoner** [comment]
Done in c64f7ce:
```
Doing epoch EQ/DQ. 
  Loading raw files for subject bad_925b.
    Epoching data (decim=5 -> sfreq=360.0 Hz).
    Computing autoreject thresholds: grad=3926 fT/cm
    Events restricted to those in params.in_numbers
    Matching trial counts and saving data to disk.
      Analysis "All": 432 epochs / condition
      Equalizing: ['std', 'ba', 'wa']
      Analysis "Individual": 19 epochs / condition
      Equalizing with sub-condition matching: ['std', 'ba', 'wa']
      Analysis "Oddball": 38 epochs / condition
      Analysis "AM": 107 epochs / condition
      Got 0 epochs for: ['s3', 's5']
      Analysis "IDs": 0, 1 epochs / condition
    Saving epochs to disk.
  (0:00:24.194)
Done

```
Requires https://github.com/LABSN/mnefun/pull/340 / now latest `main`

File: badbaby/processing/badbaby.yml [hunk 1]
```diff
@@ -487,7 +487,7 @@ epoching:
       [-1, -1, -1, 1, -1, -1, -1, -1, -1],
       [-1, -1, -1, -1, 1, 2, 3, 4, 5],
     ]
-  must_match: [[], [0, 1, 2], [0, 1, 2], [], []]  # 0-based indices
+  must_match: [[], [0, 1, 2], [0, 1], [], []]  # 0-based indices
```
Comment (ktavabi) [inline badbaby/processing/badbaby.yml:490]: @larsoner for some reason `oddball` with `[0,1]` matching doesn't behave as expected e.g. with `bad_310b` we should be getting 106 epochs/condition

```
Loading data for 824 events and 2701 original time points ...
9 bad epochs dropped
    Matching trial counts and saving data to disk.
      Analysis "All": 815 epochs / condition
      Equalizing: ['std', 'ba', 'wa']
...
      Analysis "Individual": 52 epochs / condition
      Equalizing: ['std', 'ba']
...
      Analysis "Oddball": 54, 106 epochs / condition
      Analysis "AM": 119 epochs / condition
      Analysis "IDs": 1 epochs / condition
    Saving epochs to disk.
```

**Comment by ktavabi** [comment]
I can't reproduce this on [your branch](https://github.com/larsoner/badbaby/tree/idx) 
`Analysis "Oddball": 27, 46 epochs / condition`

Which MNEFUN steps need to be rerun? `score` & `epochs`...

**Pull request merged**